src/norm.jl

```diff
@@ -0,0 +1,92 @@
+# Code adapted from Flux.jl:
+# https://github.com/FluxML/Flux.jl/blob/68ba6e4e2fa4b86e2fef8dc6d0a5d795428a6fac/src/layers/normalise.jl#L117-L206
+# License: https://github.com/FluxML/Flux.jl/blob/master/LICENSE.md
+
+mutable struct BatchNormFlow{T1,T2} <: Bijector
+    β::T1
+    logγ::T1
+    μ  # moving mean
+    σ² # moving st
+    ϵ::T2
+    momentum::T2
+    active::Bool # true when training
+end
+
+BatchNormFlow(dims::Int, container=Array; ϵ=1f-5, momentum=0.1f0) = BatchNormFlow(
+    container(zeros(Float32, dims)),
+    container(zeros(Float32, dims)),
+    zeros(Float32, dims),
+    ones(Float32, dims),
+    ϵ,
+    momentum,
+    true
+)
+
+function affinesize(x)
+    dims = length(size(x))
+    channels = size(x, dims-1)
+    affinesize = ones(Int, dims)
+    affinesize[end-1] = channels
+    return affinesize
+end
+
+logabsdetjacob(
+    t::T,
+    x;
+    σ²=reshape(t.σ², affinesize(x)...)
+) where {T<:BatchNormFlow} =  (sum(t.logγ - log.(σ² .+ t.ϵ) / 2)) .* typeof(Flux.data(x))(ones(Float32, size(x, 2))')
+
+function _transform(t::BatchNormFlow, x)
+     @assert size(x, ndims(x) - 1) == length(t.μ) "`BatchNormFlow` expected $(length(t.μ)) channels, got $(size(x, ndims(x) - 1))"
+    as = affinesize(x)
+    m = prod(size(x)[1:end-2]) * size(x)[end]
+    γ = exp.(reshape(t.logγ, as...))
+    β = reshape(t.β, as...)
+    if !t.active
+        μ = reshape(t.μ, as...)
+        σ² = reshape(t.σ², as...)
+        ϵ = t.ϵ
+    else
+        Tx = eltype(x)
+        dims = length(size(x))
+        axes = [1:dims-2; dims] # axes to reduce along (all but channels axis)
+        μ = mean(x, dims=axes)
+        σ² = sum((x .- μ) .^ 2, dims=axes) ./ m
+        ϵ = Flux.data(convert(Tx, t.ϵ))
+        # Update moving mean/std
+        mtm = Flux.data(convert(Tx, t.momentum))
+        t.μ = (1 - mtm) .* t.μ .+ mtm .* reshape(Flux.data(μ), :)
+        t.σ² = (1 - mtm) .* t.σ² .+ (mtm * m / (m - 1)) .* reshape(Flux.data(σ²), :)
+    end
+
+    x̂ = (x .- μ) ./ sqrt.(σ² .+ ϵ)
+    return (rv=γ .* x̂ .+ β, σ²=σ²)
+end
+
+(b::BatchNormFlow)(z) = _transform(b, z).rv
+
+function _forward(t::BatchNormFlow, x)
+    rv, σ² = _transform(t, x)
+    return (rv=rv, logabsdetjacob=logabsdetjacob(t, x; σ²=σ²))
```

Comment:
As mentioned above, `logabsdetjacob(t, x; σ²=σ²)` shouldn't be implemented.

I'd suggest the following:
1. Remove the entire implementation of `_forward` (this is completely redundant since it does exactly what `forward` does) and `_transform`.
2. Implement a `_compute_μ_σ²` or something that computes the `μ` and `σ²` (you're welcome to call this function `_transform` if you want). Or you can implement one for each, or whatever. Just separate out the computation of the parameters that are actually being used in different methods:)
3. `forward` should compute both `logabsdetjac` and `rv` in most efficient way; if you want to re-use the computation for `σ²` just copy-paste the implementation of `logabsdetjac` into `forward`. It's no more than a line anyways, so it's fine.

Afterwards `forward` should look something like
```julia
function forward(t::BatchNormFlow, x)
    μ, σ² = _compute_μ_σ²(t, x)  # or just put the computation in here
    γ = exp.(reshape(t.logγ, as...))
    x̂ = (x .- μ) ./ sqrt(σ² + ϵ)
    logjac = (sum(t.logγ - log.(σ² .+ t.ϵ) / 2)) .* typeof(Flux.data(x))(ones(Float32, size(x, 2))')
    return (rv=γ .* x̂, logabsdetjac=logjac)
end
```